internal/graphapi/controlextended.resolvers.go

```diff
@@ -37,96 +34,9 @@ func (r *mutationResolver) CreateControlsByClone(ctx context.Context, input *mod
-			AssessmentObjectives:   control.AssessmentObjectives,
-			ControlQuestions:       control.ControlQuestions,
-			ImplementationGuidance: control.ImplementationGuidance,
-			ExampleEvidence:        control.ExampleEvidence,
-			References:             control.References,
-			MappedControlIDs:       mappedControlIDs,
-			Status:                 &enums.ControlStatusPreparing,
-		}
-
-		if control.StandardID != "" {
-			controlInput.StandardID = &control.StandardID
-		}
-
-		if input.ProgramID != nil {
-			controlInput.ProgramIDs = []string{*input.ProgramID}
-		}
-
-		res, err := withTransactionalMutation(ctx).Control.Create().SetInput(controlInput).Save(ctx)
-		if err != nil {
-			return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
-		}
-
-		createdControlIDs = append(createdControlIDs, res.ID)
-
-		if control.Edges.Subcontrols != nil {
-			mappedControlIDs := []string{}
-			mappedControls := control.Edges.MappedControls
-			for _, mc := range mappedControls {
-				mappedControlIDs = append(mappedControlIDs, mc.ID)
-			}
-
-			// if the control has subcontrols, we need to create them as well
-			subcontrols := make([]*generated.CreateSubcontrolInput, len(control.Edges.Subcontrols))
-			for j, subcontrol := range control.Edges.Subcontrols {
-				subcontrols[j] = &generated.CreateSubcontrolInput{
-					Tags:                   subcontrol.Tags,
-					RefCode:                subcontrol.RefCode,
-					Description:            &subcontrol.Description,
-					Source:                 &subcontrol.Source,
-					ControlID:              res.ID,
-					ControlType:            &subcontrol.ControlType,
-					Category:               &subcontrol.Category,
-					CategoryID:             &subcontrol.CategoryID,
-					Subcategory:            &subcontrol.Subcategory,
-					MappedCategories:       subcontrol.MappedCategories,
-					AssessmentObjectives:   subcontrol.AssessmentObjectives,
-					ControlQuestions:       subcontrol.ControlQuestions,
-					ImplementationGuidance: subcontrol.ImplementationGuidance,
-					ExampleEvidence:        subcontrol.ExampleEvidence,
-					References:             subcontrol.References,
-					MappedControlIDs:       mappedControlIDs,
-					Status:                 &enums.ControlStatusPreparing,
-				}
-			}
-
-			if _, err := r.bulkCreateSubcontrol(ctx, subcontrols); err != nil {
-				return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
-			}
-		}
-	}
-
-	// get the created controls
-	query, err := withTransactionalMutation(ctx).Control.Query().Where(control.IDIn(createdControlIDs...)).CollectFields(ctx)
+	createdControls, err := r.cloneControls(ctx, existingControls, input.ProgramID, false)
 	if err != nil {
-		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
-	}
-
-	createdControls, err := query.All(ctx)
-	if err != nil {
-		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
+		return nil, err
```

Comment:
Resolver should return the parsed error instead of a generic error
```suggestion
		return nil, parseRequestError(err, action{action: ActionCreate, object: "control"})
```